homeassistant/components/modbus/fan.py

```diff
@@ -0,0 +1,49 @@
+"""Support for Modbus fans."""
+import logging
+from typing import Optional
+
+from homeassistant.components.fan import FanEntity
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+
+from .const import CALL_TYPE_COIL, CONF_FANS, CONF_INPUT_TYPE, MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusCoilSwitch, ModbusRegisterSwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+):
+    """Read configuration and create Modbus fan."""
+    if discovery_info is None:
+        _LOGGER.error(
+            "You're trying to init Modbus Fan in an unsupported way."
+            " Check the documentation and fix your configuration"
```

Comment:
Please add a link instead of "documentation". if you have made the right heading in your doc PR, you can #lights to send the user right to your documentation.

Reply:
OK, will do.